--- revisor/cfg.py-orig	2010-12-10 14:16:20.276631193 -0600
+++ revisor/cfg.py	2012-07-13 16:25:40.783641042 -0500
@@ -1076,6 +1076,7 @@
 
         self.pkgorder_file = ""
         self.pkgorder_style = ""
+        self.do_packageorder = False 
 
         # Use fedora-release by default
         self.release_pkgs = "^fedora-release.*$"
@@ -1147,6 +1148,9 @@
         self.samba_auth = ""
         self.nscd_auth = ""
 
+#adding option for --final , default is False
+        self.final = False
+
         # Not all options are CLI options
         try:
             if hasattr(self.plugins,"set_defaults"):
--- pungi.py	2012-07-19 14:29:17.976221559 -0500
+++ revisor/pungi.py	2012-07-16 14:23:50.800605420 -0500
@@ -303,6 +303,11 @@
         if self.cfg.install_nogr:
             buildinstall.append('--nogr')
 
+        if self.cfg.version_from in [ "RHEL6" ]:
+            #adding --final option to remove "beta" message 
+            if self.cfg.final :
+                buildinstall.append('--final')
+
         self.base.plugins.exec_hook('buildinstall_append')
 
         if hasattr(self.cfg,"buildinstall_append"):
