--- revisor/base.py-orig	2010-12-08 13:50:40.077283185 -0600
+++ revisor/base.py	2010-12-06 14:01:32.307876230 -0600
@@ -780,13 +780,13 @@
         # against the external repositories used to compose the tree, rather then
         # the tree itself
 
-        if self.cfg.version_from in [ "F9", "F10", "F11", "F12", "F13", "DEVEL" ] and mode == "installation":
+        if self.cfg.version_from in [ "F9", "F10", "F11", "F12", "F13", "DEVEL" , "RHEL6" ] and mode == "installation":
             packages_to_add = []
             required_pkgs = ['kernel']
             if self.cfg.architecture not in [ "ppc", "ppc64" ]:
                 required_pkgs.extend(['grub'])
 
-            if self.cfg.version_from not in [ "F10", "F11", "F12", "F13", "DEVEL" ]:
+            if self.cfg.version_from not in [ "F10", "F11", "F12", "F13", "DEVEL" , "RHEL6"]:
                 if self.cfg.architecture not in [ "ppc", "ppc64" ]:
                     required_pkgs.extend(['kernel-xen'])

--- revisor/pungi.py-orig	2010-12-08 13:59:11.025250380 -0600
+++ revisor/pungi.py	2010-12-02 10:40:16.865541558 -0600
@@ -275,7 +275,7 @@
         #buildinstall.append('TMPDIR=%s' % self.workdir) # TMPDIR broken in buildinstall
 
         # FIXME: Determine options from the anaconda-runtime version
-        if self.cfg.version_from in [ "F9", "F10", "F11", "F12", "F13", "DEVEL" ]:
+        if self.cfg.version_from in [ "F9", "F10", "F11", "F12", "F13", "DEVEL","RHEL6" ]:
             buildinstall.append('--debug')
 
         if self.cfg.version_from in [ "RHEL5" ]:
@@ -296,7 +296,7 @@
         buildinstall.append('"%s %s"' % (self.cfg.product_name, self.cfg.version))
 
         # FIXME: Everything < F9 needs --prodpath
-        if not self.cfg.version_from in [ "F9", "F10", "F11", "F12", "F13", "DEVEL" ]:
+        if not self.cfg.version_from in [ "F9", "F10", "F11", "F12", "F13", "DEVEL" , "RHEL6" ]:
             buildinstall.append('--prodpath')
             buildinstall.append(self.cfg.product_path)
 
@@ -317,8 +317,10 @@
             for mlist in repository_mirrorlists:
                 buildinstall.extend(['--mirrorlist', '%s' % mlist])
         else:
-            buildinstall.append('--yumconf')
-            buildinstall.append('%s' % self.cfg.main)
+            if self.cfg.version_from not in [ "RHEL6" ]:
+#RHEL6 buildinstall does not support --yumconf
+                buildinstall.append('--yumconf')
+                buildinstall.append('%s' % self.cfg.main)
 
         buildinstall.append(self.topdir)
 
