--- revisor-2.2/revisor/pungi.py.orig	2010-10-13 14:28:09.517819349 -0500
+++ revisor-2.2/revisor/pungi.py	2010-10-13 14:41:49.399702202 -0500
@@ -173,10 +173,11 @@
 
         if comps:
             # FIXME: Watch for https://bugzilla.redhat.com/show_bug.cgi?id=429509
-            if hasattr(self.cfg.yumobj.comps,"xml"):
-                ourcomps = open(comps, 'w')
-                ourcomps.write(self.cfg.yumobj.comps.xml())
-                ourcomps.close()
+#CJS this overwrites our custom comps file.  Code should be added to check first
+#            if hasattr(self.cfg.yumobj.comps,"xml"):
+#                ourcomps = open(comps, 'w')
+#                ourcomps.write(self.cfg.yumobj.comps.xml())
+#                ourcomps.close()
 
             # Let's clean up the comps file before moving anything around
             # Run the xslt filter over our comps file
@@ -670,11 +671,13 @@
     def ALL_workaround(self):
         """ Workaround the anaconda ALL "bug" """
         discinfofile = os.path.join(self.topdir, ".discinfo")
-        content = open(discinfofile, 'r').readlines()
-        #shutil.move(discinfofile, os.path.join(self.topdir, "discinfo.old"))
-        try:
-            content[content.index('ALL\n')] = '1\n'
-        except:
-            pass
+        # TJD If the .diskinfo file isn't there, it keeps going instead of failling
+        if os.path.exists(discinfofile):
+          content = open(discinfofile, 'r').readlines()
+          #shutil.move(discinfofile, os.path.join(self.topdir, "discinfo.old"))
+          try:
+              content[content.index('ALL\n')] = '1\n'
+          except:
+              pass
 
-        open(discinfofile, 'w').writelines(content)
+          open(discinfofile, 'w').writelines(content)
