--- revisor/pungi.py.original	2010-10-12 15:35:10.361910347 -0500
+++ revisor/pungi.py	2010-12-02 10:40:16.865541558 -0600
@@ -173,10 +173,11 @@
 
         if comps:
             # FIXME: Watch for https://bugzilla.redhat.com/show_bug.cgi?id=429509
-            if hasattr(self.cfg.yumobj.comps,"xml"):
-                ourcomps = open(comps, 'w')
-                ourcomps.write(self.cfg.yumobj.comps.xml())
-                ourcomps.close()
+#CJS this overwrites our custom comps file.  Code should be added to check first
+#            if hasattr(self.cfg.yumobj.comps,"xml"):
+#                ourcomps = open(comps, 'w')
+#                ourcomps.write(self.cfg.yumobj.comps.xml())
+#                ourcomps.close()
 
             # Let's clean up the comps file before moving anything around
             # Run the xslt filter over our comps file
