--- revisor-2.2/conf/revisor.conf.orig	2010-10-13 15:21:43.340667968 -0500
+++ revisor-2.2/conf/revisor.conf	2010-10-13 15:21:52.299599344 -0500
@@ -25,6 +25,29 @@
 ## Models
 ##
 
+[sl6-i386]
+description = Scientific Linux 6 for i386
+main = /etc/revisor/conf.d/revisor-sl6-i386.conf
+product_name = SL
+product_path = Packages
+architecture = i386
+version = 6
+version_from = F13
+getsource = 0
+
+
+
+[sl6-x86_64]
+description = Scientific Linux 6 for x86_64
+main = /etc/revisor/conf.d/revisor-sl6-x86_64.conf
+product_name = SL
+product_path = Packages
+architecture = x86_64
+version = 6
+version_from = F13
+getsource = 0
+
+
 [f13-i386]
 description = Fedora 13 for i386
 
